### ISSUE TITLE
Alternative names for publishers

### ISSUE THREAD

**Issue by Phyks** [dissemin/dissemin#727]
https://dissem.in/p/103878643/connecting-width-and-structure-in-knowledge-compilation is coming from https://dissem.in/b/6368/schloss-dagstuhl-leibniz-zentrum-fur-informatik but the publisher is not matched.

This is likely due to a difference in the spelling, "für" vs "fuer".

Two possible ideas here:
- Have matching with publishers names which is a little fuzzy.
- Manually consider alternative names strategy such as ö -> oe, ü -> ue, etc

**Comment by beckstefan** [comment]
In this case, we would have to do the reverse, i.e. making ue -> ü. But in fact you had to try all combinations until one succeeds, illustrated by the made up publisher "Hoetger-Institut für Jugendstilforschung". There are other languages with transcription issues, so I don't think its worth the effort.

In this particular case: Is "GmbH" and the location part of the publishers name in the publication link? All that is not found on SHERPA/RoMEO.

**Comment by a3nm** [comment]
I guess @Phyks's point is that when normalizing publisher names we could go from ü to ue so as to ensure that we only try to match strings with umlaut characters transcribed away. And yeah there's the issue of "GmbH" so it wouldn't fix the problem.

I agree that this specific hack is low-priority, but I guess we could have a more generous heuristic for publishers. Is it reasonable to assume that SHERPA knows most of them? I'm thinking about stuff like finding some publisher with a large common substring (up to some edit distance) or something like this.

**Comment by wetneb** [comment]
On the longer term, we could match SHERPA with Wikidata and use Wikidata for this matching.

Wikidata already holds many alternate names, can be edited directly, and has dedicated tools for this sort of matching.

If someone wants to have a go at this, we now have a Wikidata property to link to SHERPA: https://www.wikidata.org/wiki/Property:P6617

**Comment by beckstefan** [comment]
> I guess @Phyks's point is that when normalizing publisher names we could go from ü to ue so as to ensure that we only try to match strings with umlaut characters transcribed away. And yeah there's the issue of "GmbH" so it wouldn't fix the problem.

Yes, but SHERPA/RoMEO keeps umlauts (and for search removes the diaeresis). Or do we mean an internal matching? Then of course you're right.

**Comment by a3nm** [comment]
An idea by Pierre for approximate matching (when we get a publisher for which we can't find a matching name): use https://en.wikipedia.org/wiki/Jaro%E2%80%93Winkler_distance relative to the publishers that we already know.

**Comment by Phyks** [comment]
Naive question here, would it be better than a Levenshtein for instance?

Side note, I've successfully used https://github.com/seatgeek/fuzzywuzzy in the past, which works quite well. The main issue (as always :) is to find the right threshold value to avoid false positives.
